sat/cli/bootsys/mgmt_power.py

```diff
@@ -255,37 +256,68 @@ def do_mgmt_shutdown_power(ssh_client, username, password, excluded_ncns, ncn_sh
         ssh_client (paramiko.SSHClient): a paramiko client object.
         username (str): IPMI username to use.
         password (str): IPMI password to use.
-        excluded_ncns (set of str): The set of ncn hostnames to exclude, in
-            addition to ncn-m001, which is always excluded.
-        ncn_shutdown_timeout (int): timeout, in seconds, after which to hard
-            power off.
-        ipmi_timeout (int): timeout, in seconds, for nodes to reach desired
-            power state after IPMI power off.
+        excluded_ncns (set of str): The set of NCN hostnames to exclude.
+        ncn_shutdown_timeout (int): Timeout, in seconds, after which to hard power off.
+        ipmi_timeout (int): Timeout, in seconds, for nodes to reach desired power state after IPMI power off.
     """
-    # Ensure we do not shut down the first master node yet, as it is the node
-    # where "sat bootsys" commands are being run.
-    # TODO: Is there a better way to get the hostname of the first master node?
     try:
         other_ncns_by_role = get_and_verify_ncn_groups(excluded_ncns.union({'ncn-m001'}))
     except FatalBootsysError as err:
         LOGGER.error(f'Not proceeding with NCN power off: {err}')
         raise SystemExit(1)
 
-    other_ncns = list({ncn for role in ('managers', 'storage', 'workers')
-                       for ncn in other_ncns_by_role[role]})
-
+    # Shutdown workers
+    if other_ncns_by_role.get('workers', []):
+        worker_ncns = other_ncns_by_role['workers']
+        LOGGER.info(f'Shutting down worker NCNs: {", ".join(worker_ncns)}')
+        start_shutdown(worker_ncns, ssh_client)
+        LOGGER.info(f'Waiting up to {ncn_shutdown_timeout} seconds for worker NCNs to shut down...')
+        finish_shutdown(worker_ncns, username, password, ncn_shutdown_timeout, ipmi_timeout)
+
+    # Shutdown managers (except ncn-m001)
+    if other_ncns_by_role.get('managers', []):
+        manager_ncns = other_ncns_by_role['managers']
+        manager_ncns.remove('ncn-m001')  # Exclude ncn-m001
+        LOGGER.info(f'Shutting down manager NCNs: {", ".join(master_ncns)}')
```

Comment:
bug introduced by variable rename:

```suggestion
        LOGGER.info(f'Shutting down manager NCNs: {", ".join(manager_ncns)}')
```